ocaml/xapi/certificates.ml

```diff
@@ -387,7 +390,8 @@ let validate_certificate kind pem now private_key =
           `Msg (server_certificate_chain_invalid, []))
     )
   |> function
-    | Ok _ -> ()
+    | Ok (cert :: _)  -> cert
```

Comment:
Do you want to log that there were several entries but only the first is used?
```
| Ok [cert] ->
| Ok (cert::_) -> log ..
```

Reply:
This returns the leaf certificate if the chain. In the case of the leaf certificate we want to return it to add its metadata to the database, in the case of the chain, we don't care right now. Returning a certificate in the case of a chain is only done to keep the type checker happy.

In the future the function will change as we will most probably have to verify the whole chain every time, with the information of the leaf certificate added to the database, with a possibility to show the certs in the chain as well.